web-ui/src/streaming/import/random-data/generators.ts

```diff
@@ -0,0 +1,1067 @@
+        { type: 'SMALLINT' },
+        { type: 'INTEGER' },
+        { type: 'BIGINT' },
+        () => INTEGER_GENERATORS
+      )
+      .with({ type: 'FLOAT' }, { type: 'DOUBLE' }, () => FLOAT_GENERATORS)
+      // There aren't any good random generator libraries for arbitrary
+      // precision numbers. So we just use the same generators as FLOAT and
+      // DOUBLE for now.
+      .with({ type: 'DECIMAL' }, () => FLOAT_GENERATORS)
+      .with({ type: 'VARCHAR' }, { type: 'TEXT' }, { type: 'CHAR' }, () => STRING_GENERATORS)
+      .with({ type: 'BOOLEAN' }, () => BOOLEAN_GENERATORS)
+      .with({ type: 'TIME' }, () => TIME_GENERATORS)
+      .with({ type: 'DATE' }, () => DATE_GENERATORS)
+      .with({ type: 'TIMESTAMP' }, () => TIMESTAMP_GENERATORS)
+      .with({ type: 'ARRAY' }, () => {
+        assert(type.component !== undefined && type.component !== null)
+        return transformToArrayGenerator(type, columnTypeToRngOptions(type.component))
+      })
+      .otherwise(() => {
+        throw new Error('unsupported type for RNG')
+      })
+  )
+}
+
+// Generators for integer numbers.
+const BOOLEAN_GENERATORS: IRngGenMethod[] = [
+  {
+    title: 'Constant',
+    category: Categories.DEFAULT,
+    generator: (ct, settings) => settings?.bool_const ?? false,
+    form_fields: () => [
+      {
+        sm: 4,
+        component: BooleanSwitch,
+        props: { name: FieldNames.BOOL_CONST, label: 'Set to true' }
+      }
+    ],
+    validationSchema: () => {
+      return yup.object().shape({
+        bool_const: yup.bool()
+      })
+    }
+  },
+  {
+    title: 'Percentage',
+    category: Categories.DISTRIBUTION,
+    generator: (ct, settings) => Math.random() < (settings?.true_pct ?? 0.5),
+    form_fields: () => [
+      {
+        sm: 4,
+        component: TextField,
+        props: {
+          name: FieldNames.TRUE_PCT,
+          label: 'True [%]',
+          type: 'number',
+          inputProps: { min: 0, max: 1, step: 0.01 }
+        }
+      }
+    ],
+    validationSchema: () => {
+      return yup.object().shape({
+        true_pct: yup.number().min(0).max(1).required('Percentage required.')
+      })
+    }
+  }
+]
+
+// Generic generators for integer (and floating point) numbers.
+const NUMBER_GENERATORS: IRngGenMethod[] = [
+  {
+    title: 'Constant',
+    category: Categories.DEFAULT,
+    generator: (ct, settings) => Number(settings?.value) || 0,
+    form_fields: () => [
+      {
+        sm: 4,
+        component: TextField,
+        props: { name: FieldNames.VALUE, label: 'Constant Value', type: 'number' }
+      }
+    ],
+    validationSchema: ct => {
+      const [min, max] = typeRange(ct)
+      return yup.object().shape({
+        value: yup.number().min(min).max(max).required('Constant value is required.')
+      })
+    }
+  },
+  {
+    title: 'Normal',
+    category: Categories.DISTRIBUTION,
+    generator: (ct, props) => Math.floor(d3.randomNormal(props?.mu ?? 100, props?.sigma ?? 10)()),
```

Comment:
maybe this usage of d3 is more justified

Reply:
we only import d3-random, not all of d3